packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -157,6 +157,12 @@ class Navigator extends StatefulComponent {
 
   static void openTransaction(BuildContext context, NavigatorTransactionCallback callback) {
     NavigatorState navigator = context.ancestorStateOfType(const TypeMatcher<NavigatorState>());
+    assert(() {
+      if (navigator == null) {
+        throw new StateError('NavigatorState not found. Have you installed a Navigator in your app?');
```

Comment:
Throw a WidgetError.
For consistency with other messages, how about: "openTransaction called with a context that does not include a Navigator. The context passed to the Navigator.openTransaction() method must be that of a widget that is a descendant of a Navigator widget."